tests/v2/validation/rbac/rbac_psa_test.go

```diff
@@ -0,0 +1,336 @@
+	"github.com/stretchr/testify/suite"
+	coreV1 "k8s.io/api/core/v1"
+)
+
+const (
+	containerImage = "nginx"
+	containerName = "psanginx"
+)
+
+type PSATestSuite struct {
+	suite.Suite
+	client              *rancher.Client
+	nonAdminUser        *management.User
+	nonAdminUserClient  *rancher.Client
+	session             *session.Session
+	cluster             *management.Cluster
+	adminProject        *management.Project
+	steveAdminClient    *v1.Client
+	steveNonAdminClient *v1.Client
+	adminNamespace      *v1.SteveAPIObject
+}
+
+func (rb *PSATestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *PSATestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	rb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+
+}
+
+func (rb *PSATestSuite) ConvertNS(namespace *v1.SteveAPIObject) *coreV1.Namespace {
+	getNSSteveObject, err := rb.steveAdminClient.SteveType(namespaces.NamespaceSteveType).ByID(namespace.ID)
+	require.NoError(rb.T(), err)
+
+	namespaceObj := &coreV1.Namespace{}
+	err = v1.ConvertToK8sType(getNSSteveObject.JSONResp, namespaceObj)
+	require.NoError(rb.T(), err)
+
+	return namespaceObj
+}
+
+func (rb *PSATestSuite) ValidatePSA(role string) {
+	labels := map[string]string{
+		psaWarn:    pssPrivilegedPolicy,
+		psaEnforce: pssPrivilegedPolicy,
+		psaAudit:   pssPrivilegedPolicy,
+	}
+
+	rb.T().Logf("Validate updating the PSA labels as %v", role)
+
+	//Testcase1 Validate updating the PSA labels
+	updateNS := rb.ConvertNS(rb.adminNamespace)
+	updateNS.Labels = labels
+
+	response, err := rb.steveNonAdminClient.SteveType(namespaces.NamespaceSteveType).Update(rb.adminNamespace, updateNS)
+
+	switch role {
+	case restrictedAdmin, roleOwner:
+		require.NoError(rb.T(), err)
+		expectedLabels := getPSALabels(response, labels)
+		assert.Equal(rb.T(), labels, expectedLabels)
+	case roleMember, roleProjectReadOnly:
+		require.Error(rb.T(), err)
+		errMessage := strings.Split(err.Error(), ":")[0]
+		assert.Equal(rb.T(), "Resource type [namespace] is not updatable", errMessage)
+	case roleProjectOwner, roleProjectMember:
+		require.Error(rb.T(), err)
+		errStatus := strings.Split(err.Error(), ".")[1]
+		rgx := regexp.MustCompile(`\[(.*?)\]`)
+		errorMsg := rgx.FindStringSubmatch(errStatus)
+		assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+
+	}
+
+	rb.T().Logf("Validate deletion of the PSA labels as %v", role)
+
+	for label := range labels {
+		if strings.Contains(label, psaWarn) || strings.Contains(label, psaAudit) || strings.Contains(label, psaEnforce) {
+			delete(labels, label)
```

Comment:
Could this func name be more descriptive? I don't see this delete function.

Reply:
Agreed, maybe making this loop a helper in `rbac.go` would increase the visibility, I think this delete is for map key deletions.

Reply:
that is correct this delete if for map key deletions. Can move the loop to helper in rbac.go